summingbird-batch/src/main/scala/com/twitter/summingbird/batch/Batcher.scala

```diff
@@ -169,7 +169,13 @@ trait Batcher extends Serializable {
     dateToBatch(interval)(truncateUp)(truncateDown)
 
   def toInterval(b: BatchID): Interval[Timestamp] =
-    Intersection(InclusiveLower(earliestTimeOf(b)), ExclusiveUpper(earliestTimeOf(b.next)))
+    if (b.id < Long.MaxValue) {
```

Comment:
Does this defense make sense? I think it makes code a little bit more noisy, while don't add value in most cases (if duration != 1ms).